From 08a8e6fd1c05071a34fb8441f7531e1caf39ae36 Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Tue, 16 May 2017 22:26:39 -0400
Subject: [PATCH] Timings changes


diff --git a/src/main/java/co/aikar/timings/TimingHandler.java b/src/main/java/co/aikar/timings/TimingHandler.java
index 199789d5..c21271b3 100644
--- a/src/main/java/co/aikar/timings/TimingHandler.java
+++ b/src/main/java/co/aikar/timings/TimingHandler.java
@@ -127,7 +127,7 @@ class TimingHandler implements Timing {
         while ((last = TIMING_STACK.removeLast()) != this) {
             last.timingDepth = 0;
             if ("Minecraft".equalsIgnoreCase(last.identifier.group)) {
-                Logger.getGlobal().log(Level.SEVERE, "TIMING_STACK_CORRUPTION - Look above this for any errors and report this to Paper unless it has a plugin in the stack trace (" + last.identifier + " did not stopTiming)");
+                Logger.getGlobal().log(Level.SEVERE, "TIMING_STACK_CORRUPTION - Look above this for any errors and report this to Glowstone unless it has a plugin in the stack trace (" + last.identifier + " did not stopTiming)");
             } else {
                 Logger.getGlobal().log(Level.SEVERE, "TIMING_STACK_CORRUPTION - Report this to the plugin " + last.identifier.group + " (Look for errors above this in the logs) (" + last.identifier + " did not stopTiming)", new Throwable());
             }
diff --git a/src/main/java/co/aikar/timings/TimingsManager.java b/src/main/java/co/aikar/timings/TimingsManager.java
index ef824d70..7b8f5fb7 100644
--- a/src/main/java/co/aikar/timings/TimingsManager.java
+++ b/src/main/java/co/aikar/timings/TimingsManager.java
@@ -152,6 +152,7 @@ public final class TimingsManager {
         if (!(  server == null || pluginName == null ||
                 "minecraft".equals(pluginName) || "bukkit".equals(pluginName) ||
                 "spigot".equalsIgnoreCase(pluginName) || "paper".equals(pluginName)
+                || "glowstone".equalsIgnoreCase(pluginName)
         )) {
             plugin = server.getPluginManager().getPlugin(pluginName);
         }
-- 
2.30.2

