From b66e9534ec76f56659af73de1f789ba49424378c Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Tue, 16 May 2017 22:26:39 -0400
Subject: [PATCH] Timings changes


diff --git a/src/main/java/co/aikar/timings/TimingsExport.java b/src/main/java/co/aikar/timings/TimingsExport.java
index 65d312b0..91c39e36 100644
--- a/src/main/java/co/aikar/timings/TimingsExport.java
+++ b/src/main/java/co/aikar/timings/TimingsExport.java
@@ -204,9 +204,7 @@ class TimingsExport extends Thread {
         // Information on the users Config
 
         parent.put("config", createObject(
-            pair("spigot", mapAsJSON(Bukkit.spigot().getSpigotConfig(), null)),
-            pair("bukkit", mapAsJSON(Bukkit.spigot().getBukkitConfig(), null)),
-            pair("paper", mapAsJSON(Bukkit.spigot().getPaperConfig(), null))
+            pair("bukkit", mapAsJSON(Bukkit.spigot().getBukkitConfig(), null))
         ));
 
         new TimingsExport(listeners, parent, history).start();
diff --git a/src/main/java/co/aikar/timings/TimingsManager.java b/src/main/java/co/aikar/timings/TimingsManager.java
index ef824d70..7b8f5fb7 100644
--- a/src/main/java/co/aikar/timings/TimingsManager.java
+++ b/src/main/java/co/aikar/timings/TimingsManager.java
@@ -152,6 +152,7 @@ public final class TimingsManager {
         if (!(  server == null || pluginName == null ||
                 "minecraft".equals(pluginName) || "bukkit".equals(pluginName) ||
                 "spigot".equalsIgnoreCase(pluginName) || "paper".equals(pluginName)
+                || "glowstone".equalsIgnoreCase(pluginName)
         )) {
             plugin = server.getPluginManager().getPlugin(pluginName);
         }
-- 
2.21.0

