From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aki_ks <aki@kaysubs.de>
Date: Wed, 4 Jul 2018 15:59:22 +0200
Subject: [PATCH] Allow declaring nms classes


diff --git a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
index fed0f7a7ce7bfc41a94cba8f5844341b8d50f708..b54948f0823148622f8ddf7ba6620271a4939e82 100644
--- a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
+++ b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
@@ -179,9 +179,6 @@ public class PluginClassLoader extends URLClassLoader { // Spigot
 
     @Override
     protected Class<?> findClass(String name) throws ClassNotFoundException {
-        if (name.startsWith("org.bukkit.") || name.startsWith("net.minecraft.")) {
-            throw new ClassNotFoundException(name);
-        }
         Class<?> result = classes.get(name);
 
         if (result == null) {
