From 72f94d4c275653f7e293c3af800e44bd96febbe6 Mon Sep 17 00:00:00 2001
From: satoshinm <snmatsutake@yahoo.co.jp>
Date: Thu, 1 Jun 2017 15:13:10 -0400
Subject: [PATCH] Check for null in Sign#getFacing()


diff --git a/src/main/java/org/bukkit/material/Sign.java b/src/main/java/org/bukkit/material/Sign.java
index d33b8ed2..87ad8718 100644
--- a/src/main/java/org/bukkit/material/Sign.java
+++ b/src/main/java/org/bukkit/material/Sign.java
@@ -125,7 +125,12 @@ public class Sign extends MaterialData implements Attachable {
 
             return null;
         } else {
-            return getAttachedFace().getOppositeFace();
+            BlockFace attachedFace = getAttachedFace();
+            if (attachedFace != null) {
+                return attachedFace.getOppositeFace();
+            }
+
+            return null;
         }
     }
 
-- 
2.23.0

