From 8e47cc416c88d4ab3d34c01c8e31d053b5ecbe42 Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Thu, 24 Nov 2016 14:43:16 -0500
Subject: [PATCH] Restore Bukkit commands


diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
index bda4693..fe1106f 100644
--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
+++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
@@ -27,13 +27,50 @@ public class SimpleCommandMap implements CommandMap {
     }
 
     private void setDefaultCommands() {
+        register("bukkit", new SaveCommand());
+        register("bukkit", new SaveOnCommand());
+        register("bukkit", new SaveOffCommand());
+        register("bukkit", new StopCommand());
         register("bukkit", new ReloadCommand("reload"));
         register("bukkit", new PluginsCommand("plugins"));
         register("bukkit", new co.aikar.timings.TimingsCommand("timings")); // Spigot
     }
 
     public void setFallbackCommands() {
+        register("bukkit", new ListCommand());
+        register("bukkit", new OpCommand());
+        register("bukkit", new DeopCommand());
+        register("bukkit", new BanIpCommand());
+        register("bukkit", new PardonIpCommand());
+        register("bukkit", new BanCommand());
+        register("bukkit", new PardonCommand());
+        register("bukkit", new KickCommand());
+        register("bukkit", new TeleportCommand());
+        register("bukkit", new GiveCommand());
+        register("bukkit", new TimeCommand());
+        register("bukkit", new SayCommand());
+        register("bukkit", new WhitelistCommand());
+        register("bukkit", new TellCommand());
+        register("bukkit", new MeCommand());
+        register("bukkit", new KillCommand());
+        register("bukkit", new GameModeCommand());
         register("bukkit", new HelpCommand());
+        register("bukkit", new ExpCommand());
+        register("bukkit", new ToggleDownfallCommand());
+        register("bukkit", new BanListCommand());
+        register("bukkit", new DefaultGameModeCommand());
+        register("bukkit", new SeedCommand());
+        register("bukkit", new DifficultyCommand());
+        register("bukkit", new WeatherCommand());
+        register("bukkit", new SpawnpointCommand());
+        register("bukkit", new GameRuleCommand());
+        register("bukkit", new EnchantCommand());
+        register("bukkit", new TestForCommand());
+        register("bukkit", new EffectCommand());
+        register("bukkit", new SpreadPlayersCommand());
+        register("bukkit", new SetWorldSpawnCommand());
+        register("bukkit", new SetIdleTimeoutCommand());
+        register("bukkit", new AchievementCommand());
     }
 
     /**
-- 
2.10.2

