From b55d157ca1c2c7d8b4ed81860cf951a9bb420128 Mon Sep 17 00:00:00 2001
From: satoshinm <snmatsutake@yahoo.co.jp>
Date: Thu, 1 Jun 2017 15:13:10 -0400
Subject: [PATCH] Check for null in Sign#getFacing()


diff --git a/src/main/java/org/bukkit/material/Sign.java b/src/main/java/org/bukkit/material/Sign.java
index 3b32679b..5a02a2f4 100644
--- a/src/main/java/org/bukkit/material/Sign.java
+++ b/src/main/java/org/bukkit/material/Sign.java
@@ -131,7 +131,12 @@ public class Sign extends MaterialData implements Attachable {
 
             return null;
         } else {
-            return getAttachedFace().getOppositeFace();
+            BlockFace attachedFace = getAttachedFace();
+            if (attachedFace != null) {
+                return attachedFace.getOppositeFace();
+            }
+
+            return null;
         }
     }
 
-- 
2.24.1

