From 0afb0b0d831a417cc62ab083870ffdfdddbba8db Mon Sep 17 00:00:00 2001
From: satoshinm <snmatsutake@yahoo.co.jp>
Date: Thu, 1 Jun 2017 15:13:10 -0400
Subject: [PATCH] Check for null in Sign#getFacing()


diff --git a/src/main/java/org/bukkit/material/Sign.java b/src/main/java/org/bukkit/material/Sign.java
index d33b8ed2..96d0debb 100644
--- a/src/main/java/org/bukkit/material/Sign.java
+++ b/src/main/java/org/bukkit/material/Sign.java
@@ -3,6 +3,8 @@ package org.bukkit.material;
 import org.bukkit.block.BlockFace;
 import org.bukkit.Material;
 
+import java.util.Optional;
+
 /**
  * MaterialData for signs
  */
@@ -125,7 +127,12 @@ public class Sign extends MaterialData implements Attachable {
 
             return null;
         } else {
-            return getAttachedFace().getOppositeFace();
+            BlockFace attachedFace = getAttachedFace();
+            if (attachedFace != null) {
+                return attachedFace.getOppositeFace();
+            }
+
+            return null;
         }
     }
 
-- 
2.18.0.windows.1

