From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mastercoms <mastercoms@tuta.io>
Date: Sun, 19 Jun 2022 00:26:09 -0400
Subject: [PATCH] Define repositories for standalone builds


diff --git a/build.gradle.kts b/build.gradle.kts
index 3c4dd6eb..bc8d0fb1 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -8,6 +8,13 @@ java {
     withJavadocJar()
 }
 
+repositories {
+    mavenLocal()
+
+    maven("https://repo.glowstone.net/repository/maven-public/")
+    maven("https://repo.glowstone.net/repository/snapshots/")
+}
+
 val adventureVersion = "4.12.0"
 val apiAndDocs: Configuration by configurations.creating {
     attributes {
