From 6cda8c9df4f7455823987b01727ea63eb5cc3264 Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Sun, 20 Nov 2016 13:00:53 -0500
Subject: [PATCH] Enhance clear command


diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
index c303611..8f3f6b6 100644
--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
+++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
@@ -61,6 +61,7 @@ public class SimpleCommandMap implements CommandMap {
         register("bukkit", new DifficultyCommand());
         register("bukkit", new WeatherCommand());
         register("bukkit", new SpawnpointCommand());
+        register("bukkit", new ClearCommand());
         register("bukkit", new GameRuleCommand());
         register("bukkit", new EnchantCommand());
         register("bukkit", new TestForCommand());
diff --git a/src/main/java/org/bukkit/command/defaults/ClearCommand.java b/src/main/java/org/bukkit/command/defaults/ClearCommand.java
index 05317e1..a1491c8 100644
--- a/src/main/java/org/bukkit/command/defaults/ClearCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/ClearCommand.java
@@ -14,7 +14,6 @@ import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
-@Deprecated
 public class ClearCommand extends VanillaCommand {
     private static List<String> materials;
     static {
@@ -61,8 +60,9 @@ public class ClearCommand extends VanillaCommand {
 
             int data = args.length >= 3 ? getInteger(sender, args[2], 0) : -1;
             int count = player.getInventory().clear(id, data);
+            String endString = count == 1 ? " item" : " items";
 
-            Command.broadcastCommandMessage(sender, "Cleared the inventory of " + player.getDisplayName() + ", removing " + count + " items");
+            Command.broadcastCommandMessage(sender, "Cleared the inventory of " + player.getDisplayName() + ", removing " + count + endString);
         } else if (args.length == 0) {
             sender.sendMessage(ChatColor.RED + "Please provide a player!");
         } else {
-- 
2.10.2

