From cef4c6560800f32e75744a83bde043123c28b537 Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Sun, 12 Jun 2016 23:36:52 -0400
Subject: [PATCH] Enhance clear command


diff --git a/src/main/java/org/bukkit/command/defaults/ClearCommand.java b/src/main/java/org/bukkit/command/defaults/ClearCommand.java
index 05317e1..8382955 100644
--- a/src/main/java/org/bukkit/command/defaults/ClearCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/ClearCommand.java
@@ -61,8 +61,9 @@ public class ClearCommand extends VanillaCommand {
 
             int data = args.length >= 3 ? getInteger(sender, args[2], 0) : -1;
             int count = player.getInventory().clear(id, data);
+            String endString = count == 1 ? " item" : " items";
 
-            Command.broadcastCommandMessage(sender, "Cleared the inventory of " + player.getDisplayName() + ", removing " + count + " items");
+            Command.broadcastCommandMessage(sender, "Cleared the inventory of " + player.getDisplayName() + ", removing " + count + endString);
         } else if (args.length == 0) {
             sender.sendMessage(ChatColor.RED + "Please provide a player!");
         } else {
-- 
2.9.0.windows.1

