From 447d7bf988d741c8022fbd253eb3649809dcb654 Mon Sep 17 00:00:00 2001
From: momothereal <momothereal.mc@gmail.com>
Date: Thu, 8 Jun 2017 22:25:03 -0400
Subject: [PATCH] Fix Javadocs errors


diff --git a/src/main/java/co/aikar/timings/Timings.java b/src/main/java/co/aikar/timings/Timings.java
index 0571c9e..5c6adb6 100644
--- a/src/main/java/co/aikar/timings/Timings.java
+++ b/src/main/java/co/aikar/timings/Timings.java
@@ -143,10 +143,9 @@ public final class Timings {
     }
 
     /**
-     * Sets whether or not the Timings should monitor at Verbose level.
-     * <p/>
-     * When Verbose is disabled, high-frequency timings will not be available.
-     * Calling this will reset timing data.
+     * <p>Sets whether or not the Timings should monitor at Verbose level. Calling this will reset timing data.</p>
+     * 
+     * <p>When Verbose is disabled, high-frequency timings will not be available.</p>
      *
      * @param enabled Should high-frequency timings be reported
      */
-- 
2.10.2.windows.1

