From 9a8086a5b10c2119537c3b6a5beb10f15f1cd04b Mon Sep 17 00:00:00 2001
From: "Satoshi N. M" <snmatsutake@yahoo.co.jp>
Date: Tue, 30 May 2017 10:09:35 -0700
Subject: [PATCH] Add NPE check in Sign#getFacing()

Fixes https://github.com/GlowstoneMC/Glowstone/issues/484

diff --git a/src/main/java/org/bukkit/material/Sign.java b/src/main/java/org/bukkit/material/Sign.java
index db7822a..5116372 100644
--- a/src/main/java/org/bukkit/material/Sign.java
+++ b/src/main/java/org/bukkit/material/Sign.java
@@ -148,7 +148,7 @@ public class Sign extends MaterialData implements Attachable {
         } else {
             BlockFace attachedFace = getAttachedFace();
             if (attachedFace != null) {
-                return getAttachedFace().getOppositeFace();
+                return attachedFace.getOppositeFace();
             }
 
             return null;
-- 
2.10.2.windows.1

