From 24c238bb6020e49a4dfdc072e3e24847c92e3d72 Mon Sep 17 00:00:00 2001
From: Chris Hennick <4961925+Pr0methean@users.noreply.github.com>
Date: Mon, 29 Jan 2018 10:07:03 -0500
Subject: [PATCH] Fix NPE for Potion Effect names


diff --git a/src/main/java/org/bukkit/potion/PotionEffectTypeWrapper.java b/src/main/java/org/bukkit/potion/PotionEffectTypeWrapper.java
index 356a866f..27e7d1d1 100644
--- a/src/main/java/org/bukkit/potion/PotionEffectTypeWrapper.java
+++ b/src/main/java/org/bukkit/potion/PotionEffectTypeWrapper.java
@@ -14,7 +14,8 @@ public class PotionEffectTypeWrapper extends PotionEffectType {
 
     @Override
     public String getName() {
-        return getType().getName();
+        PotionEffectType type = getType();
+        return type == null ? "unknown effect " + getId() : type.getName();
     }
 
     /**
-- 
2.18.0.windows.1

