From ad37cf9f1a3766cc613cdee13c4df5ee5352c2a0 Mon Sep 17 00:00:00 2001
From: momothereal <momothereal.mc@gmail.com>
Date: Sat, 9 Sep 2017 23:00:00 -0400
Subject: [PATCH] Villager career API


diff --git a/src/main/java/org/bukkit/entity/Villager.java b/src/main/java/org/bukkit/entity/Villager.java
index 8fc69da..86e491b 100644
--- a/src/main/java/org/bukkit/entity/Villager.java
+++ b/src/main/java/org/bukkit/entity/Villager.java
@@ -24,11 +24,25 @@ public interface Villager extends Ageable, NPC, InventoryHolder, Merchant {
     public void setProfession(Profession profession);
 
     /**
+     * Gets the current career of this villager.
+     *
+     * @return Current career.
+     */
+    public Career getCareer();
+
+    /**
+     * Sets the new career of this villager.
+     *
+     * @param career New profession.
+     */
+    public void setCareer(Career career);
+
+    /**
      * Gets this villager's inventory.
      * <br>
      * Note that this inventory is not the Merchant inventory, rather, it is the
      * items that a villager might have collected (from harvesting crops, etc.)
-     *
+     * <p>
      * {@inheritDoc}
      */
     @Override
@@ -45,15 +59,14 @@ public interface Villager extends Ageable, NPC, InventoryHolder, Merchant {
     /**
      * Sets this villager's riches.
      *
-     * @see Villager#getRiches()
-     *
      * @param riches the new riches
+     * @see Villager#getRiches()
      */
     void setRiches(int riches);
 
     /**
      * Represents the various different Villager professions there may be.
-     * Villagers have different trading options depending on their profession,
+     * Villagers have different trading options depending on their profession and their {@link Career Career}.
      */
     public enum Profession {
         /**
@@ -103,4 +116,53 @@ public interface Villager extends Ageable, NPC, InventoryHolder, Merchant {
             return zombie;
         }
     }
+
+    /**
+     * Represents the various different Villager careers there may be.
+     */
+    public enum Career {
+        FARMER(Profession.FARMER, 1),
+        FISHERMAN(Profession.FARMER, 2),
+        SHEPHERD(Profession.FARMER, 3),
+        FLETCHER(Profession.FARMER, 4),
+
+        LIBRARIAN(Profession.LIBRARIAN, 1),
+        CARTOGRAPHER(Profession.LIBRARIAN, 2),
+
+        CLERIC(Profession.PRIEST, 1),
+
+        ARMORER(Profession.BLACKSMITH, 1),
+        WEAPON_SMITH(Profession.BLACKSMITH, 2),
+        TOOL_SMITH(Profession.BLACKSMITH, 3),
+
+        BUTCHER(Profession.BUTCHER, 1),
+        LEATHERWORKER(Profession.BUTCHER, 2),
+
+        NITWIT(Profession.NITWIT, 1);
+        private final Profession profession;
+        private final int id;
+
+        Career(Profession profession, int id) {
+            this.profession = profession;
+            this.id = id;
+        }
+
+        /**
+         * The profession that this career can be attributed from.
+         *
+         * @return the associated profession
+         */
+        public Profession getProfession() {
+            return profession;
+        }
+
+        /**
+         * The numerical ID of this career.
+         *
+         * @return the numerical ID of this career
+         */
+        public int getId() {
+            return id;
+        }
+    }
 }
-- 
2.10.2.windows.1

