From c8197f2ba6583f70201c5b073fee88f711cce501 Mon Sep 17 00:00:00 2001
From: momothereal <momothereal.mc@gmail.com>
Date: Wed, 8 Aug 2018 00:24:11 -0400
Subject: [PATCH] Add NamespacedKey->Material lookup


diff --git a/src/main/java/org/bukkit/Material.java b/src/main/java/org/bukkit/Material.java
index c628a934..836dc7c1 100644
--- a/src/main/java/org/bukkit/Material.java
+++ b/src/main/java/org/bukkit/Material.java
@@ -2940,6 +2940,7 @@ public enum Material implements Keyed {
     private final int id;
     private final Constructor<? extends MaterialData> ctor;
     private final static Map<String, Material> BY_NAME = Maps.newHashMap();
+    private final static Map<NamespacedKey, Material> BY_KEY = Maps.newHashMap();
     private final int maxStack;
     private final short durability;
     public final Class<?> data;
@@ -3858,6 +3859,17 @@ public enum Material implements Keyed {
         return BY_NAME.get(name);
     }
 
+    /**
+     * Attempts to get the Material with the given {@link NamespacedKey}.
+     *
+     * @param key Namespaced key of the material to get
+     * @return Material if found, or null
+     */
+    @Nullable
+    public static Material getMaterial(@NotNull final NamespacedKey key) {
+        return BY_KEY.get(key);
+    }
+
     /**
      * Attempts to match the Material with the given name.
      * <p>
@@ -3902,6 +3914,9 @@ public enum Material implements Keyed {
     static {
         for (Material material : values()) {
             BY_NAME.put(material.name(), material);
+            if (!material.legacy) {
+                BY_KEY.put(material.key, material);
+            }
         }
     }
 
diff --git a/src/test/java/org/bukkit/MaterialTest.java b/src/test/java/org/bukkit/MaterialTest.java
index 66f4603a..30c93102 100644
--- a/src/test/java/org/bukkit/MaterialTest.java
+++ b/src/test/java/org/bukkit/MaterialTest.java
@@ -16,7 +16,16 @@ public class MaterialTest {
 
     @Test
     public void getByNameNull() {
-        assertThat(Material.getMaterial(null), is(nullValue()));
+        assertThat(Material.getMaterial((String) null), is(nullValue()));
+    }
+
+    @Test
+    public void getByNamespacedKey() {
+        for (Material material : Material.values()) {
+            if (!material.isLegacy()) {
+                assertThat(Material.getMaterial(material.getKey()), is(material));
+            }
+        }
     }
 
     @Test
-- 
2.23.0

