From 6cc5ca4fe3445cbfcef40e072d9dbc5da84e86ea Mon Sep 17 00:00:00 2001
From: momothereal <momothereal.mc@gmail.com>
Date: Sun, 10 Sep 2017 02:07:28 -0400
Subject: [PATCH] Add entity animation API


diff --git a/src/main/java/org/bukkit/EntityAnimation.java b/src/main/java/org/bukkit/EntityAnimation.java
new file mode 100644
index 00000000..427128bb
--- /dev/null
+++ b/src/main/java/org/bukkit/EntityAnimation.java
@@ -0,0 +1,18 @@
+package org.bukkit;
+
+/**
+ * Different types of entity animations that can be played
+ */
+public enum EntityAnimation {
+    SWING_MAIN_HAND,
+
+    TAKE_DAMAGE,
+
+    LEAVE_BED,
+
+    SWING_OFF_HAND,
+
+    CRITICAL_HIT,
+
+    MAGIC_CRITICAL_HIT
+}
diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
index 4a51c519..26c89ca1 100644
--- a/src/main/java/org/bukkit/entity/LivingEntity.java
+++ b/src/main/java/org/bukkit/entity/LivingEntity.java
@@ -5,6 +5,7 @@ import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
+import org.bukkit.EntityAnimation;
 import org.bukkit.Location;
 import org.bukkit.Material;
 import org.bukkit.attribute.Attributable;
@@ -386,4 +387,11 @@ public interface LivingEntity extends Attributable, Entity, Damageable, Projecti
      */
     void setArrowsStuck(int arrows);
     // Paper end
+
+    /**
+     * Plays an animation for this entity to nearby clients.
+     *
+     * @param animation the animation to play
+     */
+    void playAnimation(EntityAnimation animation);
 }
diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index a882323d..6ae6d21f 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -6,6 +6,7 @@ import com.destroystokyo.paper.Title;
 import org.bukkit.Achievement;
 import org.bukkit.ChatColor;
 import org.bukkit.Effect;
+import org.bukkit.EntityAnimation;
 import org.bukkit.GameMode;
 import org.bukkit.Instrument;
 import org.bukkit.Location;
@@ -1715,10 +1716,18 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
     boolean hasResourcePack();
     // Paper end
 
+    /**
+     * Plays an animation to the player itself.
+     *
+     * @param animation the animation
+     */
+    void playAnimationToSelf(EntityAnimation animation);
+
     // Spigot start
     public class Spigot extends Entity.Spigot
     {
 
+
         /**
          * Gets the connection address of this player, regardless of whether it
          * has been spoofed or not.
-- 
2.16.0

