From 3e6ec21ec8e36109acf6fd9555a102c74571f59a Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Sun, 19 Jun 2016 09:11:35 -0400
Subject: [PATCH] Add more effects


diff --git a/src/main/java/org/bukkit/Effect.java b/src/main/java/org/bukkit/Effect.java
index b0f6da7..c4c2509 100644
--- a/src/main/java/org/bukkit/Effect.java
+++ b/src/main/java/org/bukkit/Effect.java
@@ -1,13 +1,12 @@
 package org.bukkit;
 
-import java.util.Map;
-
 import com.google.common.collect.Maps;
-
 import org.bukkit.block.BlockFace;
 import org.bukkit.material.MaterialData;
 import org.bukkit.potion.Potion;
 
+import java.util.Map;
+
 /**
  * A list of effects that the server is able to send to players.
  */
@@ -143,7 +142,7 @@ public enum Effect {
     FIREWORK_SHOOT(1004, Type.SOUND),
     /**
      * Particles displayed when a villager grows a plant, data
-     * is the number of particles
+     * is the number of particles, by default it is 15.
      */
     VILLAGER_PLANT_GROW(2005, Type.VISUAL, Integer.class),
     /**
@@ -340,7 +339,51 @@ public enum Effect {
      * The particles generated while sprinting a block
      * This particle requires a Material and data value so that the client can select the correct texture.
      */
-    TILE_DUST("blockdust", Type.PARTICLE, MaterialData.class)
+    TILE_DUST("blockdust", Type.PARTICLE, MaterialData.class),
+    /**
+     * The water bubble particles that appear in water, guardian laser beams and fishing.
+     */
+    BUBBLE("bubble", Type.PARTICLE),
+    /**
+     * The particle on the end of the fishing line, or the hook.
+     */
+    WAKE("wake", Type.PARTICLE),
+    /**
+     * The underwater particles.
+     */
+    UNDERWATER("suspended", Type.PARTICLE),
+    /**
+     * The crossed out circle particle that appears on barriers.
+     */
+    BARRIER("barrier", Type.PARTICLE),
+    /**
+     * Rain particles.
+     */
+    RAIN("droplet", Type.PARTICLE),
+    /**
+     * Unknown. Perhaps the particle when an item is vacuumed into the player's inventory.
+     */
+    ITEM_TAKE("take", Type.PARTICLE),
+    /**
+     * The particle used when an elder guardian appears.
+     */
+    MOB_APPEARANCE("mobappearance", Type.PARTICLE),
+    /**
+     * Damage heart when mobs are hit by a sword, axe, or shovel.
+     */
+    DAMAGE("damageIndicator", Type.PARTICLE),
+    /**
+     * The particle used for end rods and shulker bullets.
+     */
+    ROD("endRod", Type.PARTICLE),
+    /**
+     * The gray swoosh created by sword sweep attacks.
+     */
+    SWEEP("sweepAttack", Type.PARTICLE),
+    /**
+     * Dust from unstable sand and gravel.
+     */
+    DUST("fallingDust", Type.PARTICLE)
     ;
 
     private final int id;
-- 
2.9.0.windows.1

