From 7b48ce832ef9b141138df281b4171d3ee159ba88 Mon Sep 17 00:00:00 2001
From: momothereal <momothereal.mc@gmail.com>
Date: Fri, 28 Feb 2020 15:28:01 -0500
Subject: [PATCH] Add entity animation API


diff --git a/src/main/java/org/bukkit/EntityAnimation.java b/src/main/java/org/bukkit/EntityAnimation.java
new file mode 100644
index 00000000..8f04ebb9
--- /dev/null
+++ b/src/main/java/org/bukkit/EntityAnimation.java
@@ -0,0 +1,13 @@
+package org.bukkit;
+
+/**
+ * Different types of entity animations that can be played
+ */
+public enum EntityAnimation {
+    SWING_MAIN_HAND,
+    TAKE_DAMAGE,
+    LEAVE_BED,
+    SWING_OFF_HAND,
+    CRITICAL_HIT,
+    MAGIC_CRITICAL_HIT
+}
diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
index 9f0645dc..9a39bb88 100644
--- a/src/main/java/org/bukkit/entity/LivingEntity.java
+++ b/src/main/java/org/bukkit/entity/LivingEntity.java
@@ -4,6 +4,7 @@ import java.util.Collection;
 import java.util.List;
 import java.util.Set;
 import java.util.UUID;
+import org.bukkit.EntityAnimation;
 import org.bukkit.FluidCollisionMode;
 import org.bukkit.Location;
 import org.bukkit.Material;
@@ -793,6 +794,13 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
      */
     boolean isHandRaised();
 
+    /**
+     * Plays an animation for this entity to nearby clients.
+     *
+     * @param animation the animation to play
+     */
+    void playAnimation(@NotNull EntityAnimation animation);
+
     /**
      * Get entity jump state.
      * <p>
diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 34b51466..b5da9a44 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -13,6 +13,7 @@ import org.bukkit.BanList; // Paper
 import org.bukkit.Bukkit; // Paper
 import org.bukkit.DyeColor;
 import org.bukkit.Effect;
+import org.bukkit.EntityAnimation;
 import org.bukkit.GameMode;
 import org.bukkit.Instrument;
 import org.bukkit.Location;
@@ -1892,6 +1893,13 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      */
     void setPlayerProfile(@NotNull PlayerProfile profile);
 
+    /**
+     * Plays an animation to the player itself.
+     *
+     * @param animation the animation
+     */
+    void playAnimationToSelf(@NotNull EntityAnimation animation);
+
     /**
      * Returns the amount of ticks the current cooldown lasts
      *
-- 
2.30.2

