From fb991758e468da4e21104a799488967513ffb357 Mon Sep 17 00:00:00 2001
From: momothereal <momothereal.mc@gmail.com>
Date: Sun, 10 Sep 2017 02:07:28 -0400
Subject: [PATCH] Add entity animation API


diff --git a/src/main/java/org/bukkit/EntityAnimation.java b/src/main/java/org/bukkit/EntityAnimation.java
new file mode 100644
index 0000000..427128b
--- /dev/null
+++ b/src/main/java/org/bukkit/EntityAnimation.java
@@ -0,0 +1,18 @@
+package org.bukkit;
+
+/**
+ * Different types of entity animations that can be played
+ */
+public enum EntityAnimation {
+    SWING_MAIN_HAND,
+
+    TAKE_DAMAGE,
+
+    LEAVE_BED,
+
+    SWING_OFF_HAND,
+
+    CRITICAL_HIT,
+
+    MAGIC_CRITICAL_HIT
+}
diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
index be51e38..22bdf74 100644
--- a/src/main/java/org/bukkit/entity/LivingEntity.java
+++ b/src/main/java/org/bukkit/entity/LivingEntity.java
@@ -5,6 +5,7 @@ import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
+import org.bukkit.EntityAnimation;
 import org.bukkit.Location;
 import org.bukkit.Material;
 import org.bukkit.attribute.Attributable;
@@ -375,4 +376,11 @@ public interface LivingEntity extends Attributable, Entity, Damageable, Projecti
      */
     void setArrowsStuck(int arrows);
     // Paper end
+
+    /**
+     * Plays an animation for this entity to nearby clients.
+     *
+     * @param animation the animation to play
+     */
+    void playAnimation(EntityAnimation animation);
 }
diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 0c3eb03..95d4280 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -3,20 +3,7 @@ package org.bukkit.entity;
 import java.net.InetSocketAddress;
 
 import com.destroystokyo.paper.Title;
-import org.bukkit.Achievement;
-import org.bukkit.ChatColor;
-import org.bukkit.Effect;
-import org.bukkit.GameMode;
-import org.bukkit.Instrument;
-import org.bukkit.Location;
-import org.bukkit.Material;
-import org.bukkit.Note;
-import org.bukkit.OfflinePlayer;
-import org.bukkit.Particle;
-import org.bukkit.Sound;
-import org.bukkit.SoundCategory;
-import org.bukkit.Statistic;
-import org.bukkit.WeatherType;
+import org.bukkit.*;
 import org.bukkit.advancement.Advancement;
 import org.bukkit.advancement.AdvancementProgress;
 import org.bukkit.command.CommandSender;
@@ -1671,6 +1658,13 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
     boolean hasResourcePack();
     // Paper end
 
+    /**
+     * Plays an animation to the player itself.
+     *
+     * @param animation the animation
+     */
+    void playAnimationToSelf(EntityAnimation animation);
+
     // Spigot start
     public class Spigot extends Entity.Spigot
     {
-- 
2.10.2.windows.1

