From 6fe35efa915fab1633839e3bf0c80a1ff9ad15fe Mon Sep 17 00:00:00 2001
From: momothereal <momothereal.mc@gmail.com>
Date: Wed, 15 Aug 2018 11:00:21 -0400
Subject: [PATCH] Add Material#getEquipmentSlot()


diff --git a/src/main/java/org/bukkit/Material.java b/src/main/java/org/bukkit/Material.java
index e493edc5..28755076 100644
--- a/src/main/java/org/bukkit/Material.java
+++ b/src/main/java/org/bukkit/Material.java
@@ -74,6 +74,7 @@ import org.bukkit.block.data.type.Tripwire;
 import org.bukkit.block.data.type.TripwireHook;
 import org.bukkit.block.data.type.TurtleEgg;
 import org.bukkit.block.data.type.WallSign;
+import org.bukkit.inventory.EquipmentSlot;
 import org.bukkit.material.MaterialData;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
@@ -8009,4 +8010,49 @@ public enum Material implements Keyed {
             // </editor-fold>
         }
     }
+
+    @NotNull
+    public EquipmentSlot getEquipmentSlot() {
+        switch (this) {
+            // <editor-fold defaultstate="collapsed" desc="getEquipmentSlot">
+            case IRON_HELMET:
+            case LEATHER_HELMET:
+            case CHAINMAIL_HELMET:
+            case GOLDEN_HELMET:
+            case DIAMOND_HELMET:
+            case CARVED_PUMPKIN:
+            case TURTLE_HELMET:
+            case SKELETON_SKULL:
+            case WITHER_SKELETON_SKULL:
+            case ZOMBIE_HEAD:
+            case PLAYER_HEAD:
+            case CREEPER_HEAD:
+            case DRAGON_HEAD:
+                return EquipmentSlot.HEAD;
+            case IRON_CHESTPLATE:
+            case GOLDEN_CHESTPLATE:
+            case LEATHER_CHESTPLATE:
+            case CHAINMAIL_CHESTPLATE:
+            case DIAMOND_CHESTPLATE:
+            case ELYTRA:
+                return EquipmentSlot.CHEST;
+            case IRON_LEGGINGS:
+            case GOLDEN_LEGGINGS:
+            case LEATHER_LEGGINGS:
+            case CHAINMAIL_LEGGINGS:
+            case DIAMOND_LEGGINGS:
+                return EquipmentSlot.LEGS;
+            case IRON_BOOTS:
+            case GOLDEN_BOOTS:
+            case LEATHER_BOOTS:
+            case CHAINMAIL_BOOTS:
+            case DIAMOND_BOOTS:
+                return EquipmentSlot.FEET;
+            case SHIELD:
+                return EquipmentSlot.OFF_HAND;
+            default:
+                return EquipmentSlot.HAND;
+            // </editor-fold>
+        }
+    }
 }
-- 
2.24.1

