From 83f3525b244d67109ff163312e1628390724d4bc Mon Sep 17 00:00:00 2001
From: momothereal <momothereal.mc@gmail.com>
Date: Fri, 28 Feb 2020 15:28:01 -0500
Subject: [PATCH] Add entity animation API


diff --git a/src/main/java/org/bukkit/EntityAnimation.java b/src/main/java/org/bukkit/EntityAnimation.java
new file mode 100644
index 00000000..8f04ebb9
--- /dev/null
+++ b/src/main/java/org/bukkit/EntityAnimation.java
@@ -0,0 +1,13 @@
+package org.bukkit;
+
+/**
+ * Different types of entity animations that can be played
+ */
+public enum EntityAnimation {
+    SWING_MAIN_HAND,
+    TAKE_DAMAGE,
+    LEAVE_BED,
+    SWING_OFF_HAND,
+    CRITICAL_HIT,
+    MAGIC_CRITICAL_HIT
+}
diff --git a/src/main/java/org/bukkit/entity/LivingEntity.java b/src/main/java/org/bukkit/entity/LivingEntity.java
index 7f73e4b9..f44572a0 100644
--- a/src/main/java/org/bukkit/entity/LivingEntity.java
+++ b/src/main/java/org/bukkit/entity/LivingEntity.java
@@ -3,6 +3,7 @@ package org.bukkit.entity;
 import java.util.Collection;
 import java.util.List;
 import java.util.Set;
+import org.bukkit.EntityAnimation;
 import org.bukkit.FluidCollisionMode;
 import org.bukkit.Location;
 import org.bukkit.Material;
@@ -712,6 +713,13 @@ public interface LivingEntity extends Attributable, Damageable, ProjectileSource
      */
     boolean isHandRaised();
 
+    /**
+     * Plays an animation for this entity to nearby clients.
+     *
+     * @param animation the animation to play
+     */
+    void playAnimation(@NotNull EntityAnimation animation);
+
     /**
      * Get entity jump state.
      * <p>
diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 877ea4f3..0d24736e 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -9,6 +9,7 @@ import org.bukkit.BanList; // Paper
 import org.bukkit.Bukkit; // Paper
 import org.bukkit.DyeColor;
 import org.bukkit.Effect;
+import org.bukkit.EntityAnimation;
 import org.bukkit.GameMode;
 import org.bukkit.Instrument;
 import org.bukkit.Location;
@@ -1948,6 +1949,13 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      */
     void setPlayerProfile(@NotNull PlayerProfile profile);
 
+    /**
+     * Plays an animation to the player itself.
+     *
+     * @param animation the animation
+     */
+    void playAnimationToSelf(@NotNull EntityAnimation animation);
+
     /**
      * Returns the amount of ticks the current cooldown lasts
      *
-- 
2.24.1

