From b0a4943e3e4174d21de24c16d54e9e83c4834cd8 Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Fri, 25 Nov 2016 11:19:33 -0500
Subject: [PATCH] Make timings API compatible


diff --git a/src/main/java/co/aikar/timings/TimingsManager.java b/src/main/java/co/aikar/timings/TimingsManager.java
index 58ed35e..ad47a94 100644
--- a/src/main/java/co/aikar/timings/TimingsManager.java
+++ b/src/main/java/co/aikar/timings/TimingsManager.java
@@ -158,10 +158,10 @@ public final class TimingsManager {
     public static Timing getCommandTiming(String pluginName, Command command) {
         Plugin plugin = null;
         final Server server = Bukkit.getServer();
-        if (!(  server == null || pluginName == null ||
-                "minecraft".equals(pluginName) || "bukkit".equals(pluginName) ||
-                "spigot".equalsIgnoreCase(pluginName) || "paper".equals(pluginName)
-        )) {
+        if (server != null && pluginName != null &&
+                !"minecraft".equals(pluginName) && !"bukkit".equals(pluginName) &&
+                !"spigot".equalsIgnoreCase(pluginName) && !"paper".equals(pluginName) && !"glowstone".equals(pluginName)
+        ) {
             plugin = server.getPluginManager().getPlugin(pluginName);
         }
         if (plugin == null) {
-- 
2.10.2

