From 45186ff982d1d4bac6fd3275641d4026bd51a95c Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Sun, 19 Jun 2016 14:11:59 -0400
Subject: [PATCH] Enhance ageable API


diff --git a/src/main/java/org/bukkit/entity/Ageable.java b/src/main/java/org/bukkit/entity/Ageable.java
index e9fccb2..7b23f4c 100644
--- a/src/main/java/org/bukkit/entity/Ageable.java
+++ b/src/main/java/org/bukkit/entity/Ageable.java
@@ -3,7 +3,7 @@ package org.bukkit.entity;
 /**
  * Represents an entity that can age and breed.
  */
-public interface Ageable extends Creature {    
+public interface Ageable extends Creature {
     /**
      * Gets the age of this animal.
      *
@@ -49,7 +49,7 @@ public interface Ageable extends Creature {
      * @return return true if the animal is an adult
      */
     public boolean isAdult();
-    
+
     /**
      * Return the ability to breed of the animal.
      *
@@ -64,4 +64,23 @@ public interface Ageable extends Creature {
      * @param breed breedability of the animal
      */
     public void setBreed(boolean breed);
+
+    /**
+     * Creates and spawns a baby entity with the same type and location as the parent.
+     * This is a convenience method for spawning a similar entity, with the baby properties, and attributing its parent.
+     * @return The spawned baby ageable entity
+     */
+    public Ageable createBaby();
+
+    /**
+     * The parent of this baby.
+     * @return The baby's parent, null if not applicable
+     */
+    public Ageable getParent();
+
+    /**
+     * Sets the parent of this baby.
+     * @param parent The new baby's parent
+     */
+    public void setParent(Ageable parent);
 }
-- 
2.9.0.windows.1

