From f61008d6c324b676b6566e1b487d774d0c08e06f Mon Sep 17 00:00:00 2001
From: aki_ks <aki@kaysubs.de>
Date: Wed, 4 Jul 2018 15:59:22 +0200
Subject: [PATCH] Allow declaring nms classes


diff --git a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
index 84a40825..fdf89113 100644
--- a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
+++ b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
@@ -105,9 +105,6 @@ public class PluginClassLoader extends URLClassLoader { // Spigot
     }
 
     Class<?> findClass(String name, boolean checkGlobal) throws ClassNotFoundException {
-        if (name.startsWith("org.bukkit.") || name.startsWith("net.minecraft.")) {
-            throw new ClassNotFoundException(name);
-        }
         Class<?> result = classes.get(name);
 
         if (result == null) {
-- 
2.18.0

