From 49233b401f2e8be03b6479a5046b2b94e2a99b61 Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Fri, 25 Nov 2016 12:30:51 -0500
Subject: [PATCH] Enhance damageable API


diff --git a/src/main/java/org/bukkit/entity/Damageable.java b/src/main/java/org/bukkit/entity/Damageable.java
index d0599cf..1d28e1e 100644
--- a/src/main/java/org/bukkit/entity/Damageable.java
+++ b/src/main/java/org/bukkit/entity/Damageable.java
@@ -1,5 +1,7 @@
 package org.bukkit.entity;
 
+import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
+
 /**
  * Represents an {@link Entity} that has health and can take damage.
  */
@@ -42,6 +44,25 @@ public interface Damageable extends Entity {
     void _INVALID_damage(int amount, Entity source);
 
     /**
+     * Deals the given amount of damage to this entity, from a specified
+     * cause.
+     *
+     * @param amount Amount of damage to deal
+     * @param cause  Cause of the damage
+     */
+    void damage(double amount, DamageCause cause);
+
+    /**
+     * Deals the given amount of damage to this entity, from a specified
+     * cause.
+     *
+     * @param amount Amount of damage to deal
+     * @param source Entity which to attribute this damage from
+     * @param cause  Cause of the damage
+     */
+    void damage(double amount, Entity source, DamageCause cause);
+
+    /**
      * Gets the entity's health from 0 to {@link #getMaxHealth()}, where 0 is dead.
      *
      * @return Health represented from 0 to max
-- 
2.10.2

