From 6f0b9f0171dfbc15c4501f41626670414fb74cab Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Fri, 25 Nov 2016 12:32:08 -0500
Subject: [PATCH] Enhance title API


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index bf174445..b75a0eec 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -568,6 +568,13 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
     // Paper end
 
     /**
+     * Gets the title currently set on the player
+     *
+     * @return the current title
+     */
+    public Title getTitle();
+
+    /**
      * Forces an update of the player's entire inventory.
      *
      */
-- 
2.11.0

