From 5aeae8ade389c0d3af1cc2580cbe23b375f9427f Mon Sep 17 00:00:00 2001
From: mastercoms <circuitsoft@tuta.io>
Date: Sun, 19 Jun 2016 14:25:50 -0400
Subject: [PATCH] Add SpawnMeta


diff --git a/src/main/java/org/bukkit/inventory/meta/SpawnMeta.java b/src/main/java/org/bukkit/inventory/meta/SpawnMeta.java
new file mode 100644
index 0000000..e83a7c4
--- /dev/null
+++ b/src/main/java/org/bukkit/inventory/meta/SpawnMeta.java
@@ -0,0 +1,29 @@
+package org.bukkit.inventory.meta;
+
+import org.bukkit.entity.EntityType;
+
+public interface SpawnMeta extends ItemMeta {
+
+    /**
+     * Checks if an Entity type is attributed to the spawn egg.
+     *
+     * @return true if an Entity type is attributed to the spawn egg
+     */
+    boolean hasEntityType();
+
+    /**
+     * Gets the Entity type attributed to the spawn egg.
+     *
+     * @return The entity type attributed to the spawn egg
+     */
+    EntityType getEntityType();
+
+    /**
+     * Sets the Entity type attributed to the spawn egg.
+     *
+     * @param type The Entity type to attribute to the spawn egg
+     */
+    void setEntityType(EntityType type);
+
+    SpawnMeta clone();
+}
diff --git a/src/main/java/org/bukkit/material/SpawnEgg.java b/src/main/java/org/bukkit/material/SpawnEgg.java
index 596d08a..bc2bb46 100644
--- a/src/main/java/org/bukkit/material/SpawnEgg.java
+++ b/src/main/java/org/bukkit/material/SpawnEgg.java
@@ -2,7 +2,7 @@ package org.bukkit.material;
 
 import org.bukkit.Material;
 import org.bukkit.entity.EntityType;
-import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.SpawnMeta;
 
 /**
  * Represents a spawn egg that can be used to spawn mobs
@@ -41,7 +41,7 @@ public class SpawnEgg extends MaterialData {
      * Get the type of entity this egg will spawn.
      *
      * @return The entity type.
-     * @deprecated This is now stored in {@link ItemMeta}. See SPIGOT-1592.
+     * @deprecated See {@link SpawnMeta}.
      */
     @Deprecated
     public EntityType getSpawnedType() {
@@ -52,7 +52,7 @@ public class SpawnEgg extends MaterialData {
      * Set the type of entity this egg will spawn.
      *
      * @param type The entity type.
-     * @deprecated This is now stored in {@link ItemMeta}. See SPIGOT-1592.
+     * @deprecated See {@link SpawnMeta}.
      */
     @Deprecated
     public void setSpawnedType(EntityType type) {
diff --git a/src/main/java/org/bukkit/plugin/PluginManager.java b/src/main/java/org/bukkit/plugin/PluginManager.java
index e5638d5..b2b6918 100644
--- a/src/main/java/org/bukkit/plugin/PluginManager.java
+++ b/src/main/java/org/bukkit/plugin/PluginManager.java
@@ -1,14 +1,14 @@
 package org.bukkit.plugin;
 
-import java.io.File;
-import java.util.Set;
-
 import org.bukkit.event.Event;
 import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
 import org.bukkit.permissions.Permissible;
 import org.bukkit.permissions.Permission;
 
+import java.io.File;
+import java.util.Set;
+
 /**
  * Handles all plugin management from the Server
  */
-- 
2.9.0.windows.1

