From e64f17b6a9219d74f73e7a50aa6d0697bb8aa27b Mon Sep 17 00:00:00 2001
From: momothereal <momothereal.mc@gmail.com>
Date: Fri, 25 Nov 2016 12:40:15 -0500
Subject: [PATCH] Enhance SpawnEggMeta


diff --git a/src/main/java/org/bukkit/inventory/meta/SpawnEggMeta.java b/src/main/java/org/bukkit/inventory/meta/SpawnEggMeta.java
index b89802d1..b3ccf6e3 100644
--- a/src/main/java/org/bukkit/inventory/meta/SpawnEggMeta.java
+++ b/src/main/java/org/bukkit/inventory/meta/SpawnEggMeta.java
@@ -4,11 +4,18 @@ import org.bukkit.Material;
 import org.bukkit.entity.EntityType;
 
 /**
- * Represents a {@link Material#MONSTER_EGG} and it's spawned type.
+ * Represents a {@link Material#MONSTER_EGG} and its spawned type.
  */
 public interface SpawnEggMeta extends ItemMeta {
 
     /**
+     * Checks if an Entity type is attributed to the spawn egg.
+     *
+     * @return true if an Entity type is attributed to the spawn egg
+     */
+    boolean hasSpawnedType();
+
+    /**
      * Get the type of entity this egg will spawn.
      *
      * @return The entity type. May be null for implementation specific default.
-- 
2.11.0

