From e8eb6261a467a8da5e53ad2837180aec2bd23470 Mon Sep 17 00:00:00 2001
From: momothereal <momothereal.mc@gmail.com>
Date: Fri, 25 Nov 2016 12:40:15 -0500
Subject: [PATCH] Add SpawnMeta


diff --git a/src/main/java/org/bukkit/inventory/meta/SpawnMeta.java b/src/main/java/org/bukkit/inventory/meta/SpawnMeta.java
new file mode 100644
index 0000000..e83a7c4
--- /dev/null
+++ b/src/main/java/org/bukkit/inventory/meta/SpawnMeta.java
@@ -0,0 +1,29 @@
+package org.bukkit.inventory.meta;
+
+import org.bukkit.entity.EntityType;
+
+public interface SpawnMeta extends ItemMeta {
+
+    /**
+     * Checks if an Entity type is attributed to the spawn egg.
+     *
+     * @return true if an Entity type is attributed to the spawn egg
+     */
+    boolean hasEntityType();
+
+    /**
+     * Gets the Entity type attributed to the spawn egg.
+     *
+     * @return The entity type attributed to the spawn egg
+     */
+    EntityType getEntityType();
+
+    /**
+     * Sets the Entity type attributed to the spawn egg.
+     *
+     * @param type The Entity type to attribute to the spawn egg
+     */
+    void setEntityType(EntityType type);
+
+    SpawnMeta clone();
+}
diff --git a/src/main/java/org/bukkit/material/SpawnEgg.java b/src/main/java/org/bukkit/material/SpawnEgg.java
index 596d08a..8f05cbd 100644
--- a/src/main/java/org/bukkit/material/SpawnEgg.java
+++ b/src/main/java/org/bukkit/material/SpawnEgg.java
@@ -2,7 +2,7 @@ package org.bukkit.material;
 
 import org.bukkit.Material;
 import org.bukkit.entity.EntityType;
-import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.SpawnMeta;
 
 /**
  * Represents a spawn egg that can be used to spawn mobs
@@ -41,7 +41,7 @@ public class SpawnEgg extends MaterialData {
      * Get the type of entity this egg will spawn.
      *
      * @return The entity type.
-     * @deprecated This is now stored in {@link ItemMeta}. See SPIGOT-1592.
+     * @deprecated This is now stored in {@link SpawnMeta}.
      */
     @Deprecated
     public EntityType getSpawnedType() {
@@ -52,7 +52,7 @@ public class SpawnEgg extends MaterialData {
      * Set the type of entity this egg will spawn.
      *
      * @param type The entity type.
-     * @deprecated This is now stored in {@link ItemMeta}. See SPIGOT-1592.
+     * @deprecated This is now stored in {@link SpawnMeta}.
      */
     @Deprecated
     public void setSpawnedType(EntityType type) {
-- 
2.10.2

