From 437499212caaf557cf4c5bbe17763e04d3d13e29 Mon Sep 17 00:00:00 2001
From: Momo <momo@momoperes.ca>
Date: Fri, 12 Aug 2016 13:27:20 -0400
Subject: [PATCH] Fix setblock command


diff --git a/src/main/java/org/bukkit/command/defaults/SetBlockCommand.java b/src/main/java/org/bukkit/command/defaults/SetBlockCommand.java
index 9ebd30c..c87e4c8 100644
--- a/src/main/java/org/bukkit/command/defaults/SetBlockCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/SetBlockCommand.java
@@ -1,11 +1,11 @@
 package org.bukkit.command.defaults;
 
-import org.bukkit.Bukkit;
-import org.bukkit.ChatColor;
-import org.bukkit.Material;
-import org.bukkit.World;
+import org.bukkit.*;
 import org.bukkit.block.Block;
+import org.bukkit.command.BlockCommandSender;
 import org.bukkit.command.CommandSender;
+import org.bukkit.command.CommandUtils;
+import org.bukkit.entity.Entity;
 import org.bukkit.entity.Player;
 
 public class SetBlockCommand extends VanillaCommand {
@@ -29,10 +29,16 @@ public class SetBlockCommand extends VanillaCommand {
             sender.sendMessage(ChatColor.RED + "Usage: " + usageMessage);
             return false;
         }
+        Location base = null;
+        if (sender instanceof BlockCommandSender) {
+            base = ((BlockCommandSender) sender).getBlock().getLocation();
+        } else if (sender instanceof Entity) {
+            base = ((Entity) sender).getLocation();
+        }
+
+        String xP = args[0], yP = args[1], zP = args[2];
+        Location location = CommandUtils.getLocation(base, xP, yP, zP);
 
-        int x = getInteger(sender, args[0], 0); // TODO: support relative '~' notation like in TeleportCommand
-        int y = getInteger(sender, args[1], 0);
-        int z = getInteger(sender, args[2], 0);
         String tileName = args[3];
 
         byte dataValue = 0;
@@ -69,9 +75,9 @@ public class SetBlockCommand extends VanillaCommand {
             return false;
         }
 
-        Block block = world.getBlockAt(x, y, z);
+        Block block = world.getBlockAt(location);
         if (block == null) {
-            sender.sendMessage(ChatColor.RED + "No block found at (" + x + "," + y + "," + z + ")");
+            sender.sendMessage(ChatColor.RED + "No block found at (" + location.getBlockX() + "," + location.getBlockY() + "," + location.getBlockZ() + ")");
             return false;
         }
 
@@ -90,13 +96,15 @@ public class SetBlockCommand extends VanillaCommand {
 
         boolean applyPhysics = true;
 
-        block.setTypeIdAndData(material.getId(), dataValue, applyPhysics);
-        //block.setType(material);
-        //block.setData(dataValue);
+        //block.setTypeIdAndData(material.getId(), dataValue, applyPhysics);
+        block.setType(material);
+        block.setData(dataValue);
 
         if (oldBlockHandling == OldBlockHandling.DESTROY) {
+            throw new UnsupportedOperationException("Destroying blocks not supported.");
             // TODO: destory old block, play breaking sound, drop item as if broken by player
         }
+        sender.sendMessage("Block placed.");
 
         return true;
     }
-- 
1.9.5.msysgit.1

